### DISCUSSION TITLE
Restore Backup File in UNRAID

### DISCUSSION THREAD

**Discussion by droideggs** [teslamate-org/teslamate#4248] [General]
I am trying to perform a restore of a backup I made using the instructions provided in the documentation, but restoring this file in UNRAID seems to not be possible. Does anyone know the steps/commands needed to restore the backup? 

I do have pgadmin4 app i grabbed from the community appstore, and was able to connect to the teslamate db. when trying to restore the file however, I get an error saying that the file is in a text-format. 
![image](https://github.com/user-attachments/assets/4357e6db-51c5-48f4-80d4-f5fc901d99c4)

Is there a command based method to restore the backup file on UNRAID? Thanks in advance

**Comment by brianmay** [comment]
Disclaimer: this is new to me, so take care...

Looking at the information https://www.pgadmin.org/docs/pgadmin4/latest/restore_dialog.html it looks like this needs the backup to be in tar format.

`pg_dump` will create a tar format with the `--format=t` option.

(I had to do this recently when copying a database from one server to another server because the newer psql seemed to have problems reading the older sql dump)

**Reply by droideggs** [reply]
sorry im a complete newb, would you be able to write out the exact command to get this done? i ended up doing a psql command on my MacOS remoting into the unraid server

`psql -U teslamate -p 5432 -h 192.168.68.xx -d teslamate < teslamate.bck`

but i got all kinds of errors and it corrupted the database

**Reply by brianmay** [reply]
You haven't said what errors you got.

But my guess is that you are exporting from one version of postgresql and importing to another. As I said above, this doesn't always work when using the text based SQL dumps.

If you have the original database still, it might be best to dump the database in tar format.
